src/components/StageForm.js

```diff
@@ -1,78 +1,98 @@
-    console.log(this.validatorTypes)
+    // Setup this.state
+    this.state = this.setupState(
+      props.getStore(),
+      Object.keys(props.form)
+    );
 
-    this.getValidatorData = this.getValidatorData.bind(this);
-    this.renderHelpText = this.renderHelpText.bind(this);
+    this.onChange = this.onChange.bind(this);
     this.isValidated = this.isValidated.bind(this);
   }
 
-  isValidated() {
-    return new Promise((resolve, reject) => {
-      this.props.validate((error) => {
-        if (error) {
-          reject();
-          return;
-        }
-        resolve();
-      });
+  setupState(store, keys) {
+    let newState = {};
+
+    // Create "error" keys for each form value, and set it to null
+    keys.forEach((key) => {
+      let errorName = key + '_error';
+
+      newState[key] = store[key];
+      newState[errorName] = null;
     });
-  }
 
-  getValidatorData() {
-    return this.makeValidationObject(this.props.form, (key) => this.props.getStore()[key])
+    return newState;
   }
 
-  makeValidationObject(key, valueCb) {
-    return Object.entries(this.props.form)
-      .map(([key, value]) => {
-        return [key, valueCb(key)];
-      })
-      .filter(([key, value]) => value !== null && value !== undefined)
-      .reduce((result, [key, value]) => {
-        result[key] = value;
-        return result
-      }, {});
-  }
+  isValidated() {
+    let isDataValid = true;
+    let newState = {};
+    let newStore = {};
 
-  handleChange(event) {
-    const target = event.target;
+    // Iterate over props.form fields and update field's error status
+    Object.keys(this.props.form).forEach((key) => {
+      let schema = this.props.form[key].validation;
+      let value = this.state[key];
+      let validation = Joi.validate(value, schema);
+      let errorName = key + '_error';
 
-    this.props.updateStore({
-      [target.name]: target.type === 'checkbox'
-        ? target.checked
-        : target.value
+      if (validation.error !== null) {
+        isDataValid = false;
+        // set error message for given field property
+        newState[errorName] = validation.error.details[0].message;
+      } else {
+        // set null to error message value for given field property
+        newState[errorName] = null;
+        newStore[key] = value;
+      }
     });
+    // Update state and local store
+    this.setState(newState);
+    if (isDataValid) {
+      this.props.updateStore(newStore);
+    }
+
+    return isDataValid;
   }
 
-  renderHelpText(message, id) {
-    return (<div className="val-err-tooltip" key={id}><span>{message}</span></div>);
+  onChange(e) {
+    let newState = {};
+
+    newState[e.target.name] = e.target.value;
+    this.setState(newState);
   }
 
   render() {
     return (
       <Form>
         {Object.entries(this.props.form).map(([key, value]) => {
+          let errorName = key + "_error";
```

Comment:
This constitutes a small bit of repeated code... might be clearer to introduce a "makeErrorKey" function
`function makeErrorKey(key) {
  return key + "_error";
}`
and use it in setupState(), handleChange() and render() to emphasize they are all related.